crates/matrix-sdk-ui/src/timeline/read_receipts.rs

```diff
@@ -74,11 +97,9 @@ impl ReadReceipts {
         timeline_items: &mut ObservableVectorTransaction<'_, Arc<TimelineItem>>,
     ) {
         // Get old receipt.
-        let old_receipt = self
-            .users_read_receipts
-            .get(new_receipt.user_id)
-            .and_then(|receipts| receipts.get(&new_receipt.receipt_type));
+        let old_receipt = self.get_latest(new_receipt.user_id, &new_receipt.receipt_type).cloned();
```

Comment:
We didn't clone here before, so don't need to do so now, right?

Reply:
Ah, it's because of the new `remove_event_for_user_id(&mut self)` creating an immutable vs mutable borrow conflict. Since I kind of like this new API though, I added a clone of a captured `Option<&OwnedEventId>` further below.